valhalla/baldr/double_bucket_queue.h

```diff
@@ -33,13 +39,35 @@ template <typename label_t> class DoubleBucketQueue final {
    * @param range      Cost range for low-level buckets.
    * @param bucketsize Bucket size (range of costs within same bucket).
    *                   Must be an integer value.
-   * @param labelcost  Functor to get a cost given a label index.
+   * @param labelcontainer  Container of labels with sortcosts.
    */
   DoubleBucketQueue(const float mincost,
                     const float range,
                     const uint32_t bucketsize,
-                    const std::vector<label_t>& labelcontainer)
-      : labelcontainer_(labelcontainer) {
+                    const std::vector<label_t>* labelcontainer) {
+    reuse(mincost, range, bucketsize, labelcontainer);
+  }
+
+  DoubleBucketQueue(DoubleBucketQueue&&) = default;
+  DoubleBucketQueue& operator=(DoubleBucketQueue&&) = default;
+  DoubleBucketQueue(const DoubleBucketQueue&) = delete;
+  DoubleBucketQueue& operator=(const DoubleBucketQueue&) = delete;
+
+  /**
+   * The same as c-tor, but without buffers reallocation. Before call this
+   * method you should clean up the current state (call `clear`).
+   * @param mincost    Minimum cost. Used to create the initial range for
+   *                   bucket sorting.
+   * @param range      Cost range for low-level buckets.
+   * @param bucketsize Bucket size (range of costs within same bucket).
+   *                   Must be an integer value.
+   * @param labelcontainer  Container of labels with sortcosts.
+   */
+  void reuse(const float mincost,
```

Comment:
maybe we should call this `reserve` to be more stl like

Reply:
hm, interesting idea. the only difference this is `must-have` function to use the queue. in stl it's not necessary to call `reserve` before usage

Reply:
no worries just a suggestion